src/backend/base/langflow/components/git/gitfile.py

```diff
@@ -0,0 +1,186 @@
+                await check_file.read()
+                return False
+        except UnicodeDecodeError:
+            return True
+
+    async def get_repository_files(self, repo_path: str) -> list[str]:
+        file_list = []
+        path = anyio.Path(repo_path)  # Convert str to anyio.Path explicitly
+        async for entry in path.rglob("*"):
+            if await entry.is_file():
+                relative_path = os.path.relpath(str(entry), str(path))
+                if not relative_path.startswith(".git"):
+                    file_list.append(relative_path)
+        return sorted(file_list)
+
+    async def get_branches(self, repo_url: str) -> list[str]:
+        try:
+            async with self.temp_git_repo() as temp_dir:
+                repo = await asyncio.get_event_loop().run_in_executor(
+                    None, lambda: git.Repo.clone_from(repo_url, temp_dir, no_checkout=True)
+                )
+                await asyncio.get_event_loop().run_in_executor(None, lambda: repo.remote().fetch())
+                branches = []
+
+                for ref in repo.remote().refs:
+                    if ref.name == "origin/HEAD":
+                        continue
+                    branch_name = ref.name.split("/")[-1]
+                    if branch_name not in branches:
+                        branches.append(branch_name)
+
+                return sorted(branches) if branches else ["main", "master"]
+        except git.GitCommandError as e:
+            self.log(f"Error getting branches: {e!s}")
+            return ["main", "master"]
+
+    async def update_build_config(self, build_config: dict, _field_value: str, field_name: str | None = None) -> dict:
+        if not self.repository_url:
+            build_config["branch"]["options"] = ["Enter repository URL first"]
+            build_config["branch"]["value"] = "Enter repository URL first"
+            build_config["selected_files"]["options"] = ["Select branch first"]
+            build_config["selected_files"]["value"] = []
+            return build_config
+
+        if field_name in {"repository_url", "branch"}:
+            branches = await self.get_branches(self.repository_url)
+            build_config["branch"]["options"] = branches
+            if field_name == "repository_url":
+                build_config["branch"]["value"] = branches[0] if branches else "main"
+
+        if (
+            self.branch
+            and self.branch not in ["Enter repository URL first"]
+            and field_name in {"repository_url", "branch", "selected_files"}
+        ):
+            try:
+                async with self.temp_git_repo() as temp_dir:
+                    await asyncio.get_event_loop().run_in_executor(
+                        None,
+                        lambda: git.Repo.clone_from(
+                            self.repository_url, temp_dir, branch=self.branch, depth=1, single_branch=True
+                        ),
+                    )
```

Comment:
This will not work. Use this instead:
```suggestion
                    await asyncio.to_thread(
                git.Repo.clone_from,
                self.repository_url,
                temp_dir,
                branch=self.branch,
                depth=1,
                single_branch=True,
            )
```